code/model/ListFilterDateRange.php

```diff
@@ -88,8 +88,9 @@ public function applyFilter(SS_List $list, array $data) {
 	 * @return SS_List
 	 */
 	public function applyDateRange(SS_List $list, $start, $end) {
+		// get both date fields, or a single field
 		$startDateField = $this->StartDateField;
-		$endDateField = $this->EndDateField;
+		$endDateField = $this->EndDateField ? $this->EndDateField : $startDateField;
```

Comment:
Just an FYI, you can do a short ternary operation here which is the equivalent to your code:

`$endDateField = $this->EndDateField ?: $startDateField;`

Reply:
True! Done.